### PR TITLE
Performance and Network enhancements

### PR DESCRIPTION
**Pull request by chrispable**
This PR removes my debug device from the p3io code and makes it less chatty in the log reducing log spam and disk thrashing and io bottlenecking. The statements won't help anyone anyway and were only meant to be temporary. I left them there so people could understand what the code was doing, however, in case someone else wanted to write an IO driver for it.

Also adds the ability to add a defined timeout to ezsockets, abstracts the http logic from screenpackages, and allows it to be used to, for example, broadcast the current song being played. I also added the ability to thread these requests so the game thread isn't lagged. The end result will be nice for streamers and allow enhanced functionality in the future (I'll be making a score broadcaster as well, great for tournaments and conventions).

The last fix has to do with USB drive mounting. Every single tutorial, including, for example, this one: http://www.hackmycab.com/?portfolio=configuring-usbs shows USB mount points for windows using a trailing slash or backslash. The current mount code for windows now doesn't like the trailing slash. This change allows the legacy behavior in the configuration file by simply stripping slashes off the end until it doesn't see a slash.

P.S. Visual studio's auto formatter makes me weep. I'm very sorry for the BS it created in generating the diff. I used to use notepad2 and just write in there. 

Update:
I ended up painfully reverting all the msvc formatter changes so they would display cleanly. I also ended up sending both machine and profile guid to the score submission system.

In ezsockets I added the ability to actually affect the real socket blocking behavior as well as added function analogs that do not use select (for the rare instances you don't want that).

I corrected url encoding bugs from the function from screenpackages and allowed it to work for UTF8 strings as well.

### CODE REVIEW

File: src/ezsockets.cpp [hunk 1]
```diff
@@ -133,6 +139,21 @@ bool EzSockets::listen()
...
 #endif
 
+void EzSockets::setBlocking(bool b)
+{
+	#if defined(WIN32)
+	u_long iMode = 0;
+	if (!b)	iMode=1;
+	ioctlsocket(sock, FIONBIO, &iMode);
+	#else
+	const int flags = fcntl(socket, F_GETFL, 0);
```
Comment (spigwitmer) [inline src/ezsockets.cpp]: did you mean for this to be `sock` rather than `socket`?

File: src/ezsockets.cpp [hunk 2]
```diff
@@ -133,6 +139,21 @@ bool EzSockets::listen()
...
+    if ((flags & O_NONBLOCK) && !b) { return; }
+    if (!(flags & O_NONBLOCK) && b) { return; }
+    fcntl(sock, F_SETFL, b ? flags ^ O_NONBLOCK : flags | O_NONBLOCK));
```
Comment (spigwitmer) [inline src/ezsockets.cpp]: missing beginning parenthesis for ternary?

File: src/HTTPHelper.cpp [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+					sAddress = "/" + sAddress;
+
+				wSocket.close();
+				wSocket.create();
+				wSocket.setTimeout(0,250000); //acceptable timeout
+				wSocket.setBlocking(true);
+				
+				if( !wSocket.connect( Server, (short) Port ) )
+				{
+					LOG->Info("HTTPHelper::SubmitPostRequest failed to connect to %s:%d ", Server,Port);
```
Comment (spigwitmer) [inline src/HTTPHelper.cpp]: Server.c_str()

File: src/HTTPHelper.cpp [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+					sBUFFER.append( res, iSize );
+					BytesGot += iSize;
+					if( iSize < HTTP_CHUNK_SIZE )
+						break;
+				}
+				wSocket.close();
+		}
+		else
+		{
+			LOG->Debug("HTTPHelper::SubmitPostRequest could not parse %s ", URL);
```
Comment (spigwitmer) [inline src/HTTPHelper.cpp]: URL.c_str()

File: src/ezsockets.cpp (hunk 1 above)
Comment (chrispable) [inline src/ezsockets.cpp]: Yeah that's a typo. I hadn't tested Linux yet. Thanks for catching it.

File: src/ezsockets.cpp (hunk 2 above)
Comment (chrispable) [inline src/ezsockets.cpp]: That line should be fine but I'll see if GCC barks at me

File: src/HTTPHelper.cpp (hunk 4 above)
Comment (chrispable) [inline src/HTTPHelper.cpp]: Yup. Messed this up. This I wanted to be info instead of debug as well. Will get that fixed up in a few hours.

**Comment by chrispable** [comment]
Thanks for the spot check. I'll fix those. I actually didn't realize my changes would keep piling on to the same PR from the branch and your seeing a few things that were under active development. On the windows side (aside from the error logging stuff, which I didn't hit, it worked great and everything is feature complete. Was going to upload the source and test Linux on a cloud box later today. I'll reply when ive tested Linux as well so you don't have to go through the pain of that :)

**Comment by chrispable** [comment]
Confirmed linux compilation is OK with the above commits!

**Comment by spigwitmer** [comment]
looks good, thanks for addressing the changes there. merging.

**Pull request merged**